src/server/gc_worker/gc_manager.rs

```diff
@@ -51,18 +52,27 @@ pub struct AutoGcConfig<S: GcSafePointProvider, R: RegionInfoProvider> {
     /// This will be called when a round of GC has finished and goes back to
     /// idle state. This field is for test purpose.
     pub post_a_round_of_gc: Option<Box<dyn Fn() + Send>>,
+
+    /// Used to save tablet factory of TiKV instance
+    pub tablet_factory: Option<Arc<dyn TabletFactory<RocksEngine> + Send + Sync>>,
```

Comment:
Drop the `Option`, there should always be an available table factory.

Reply:
I have to keep it as `Option` to support legacy unit test cases, where it's hard to construct a factory.

Reply:
Actually, there are many similar cases where using Option can ease the adaptation of legacy unit tests when making modifications for v2, and I am also interested in whether this is a good method for this.

Reply:
Since tablet_factory is definitely not None at runtime, I removed the Option.